routing/payment_session.go

```diff
@@ -363,3 +366,63 @@ func (p *paymentSession) RequestRoute(maxAmt, feeLimit lnwire.MilliSatoshi,
 		return route, err
 	}
 }
+
+// UpdateAdditionalEdge looks up a channel edge policy using the ChannelID from
+// the msg. If the policy is found, it then tries to update it with the
+// supplied value. It returns a boolean to indicate whether the update is meant
+// for additional edges.
+func (p *paymentSession) UpdateAdditionalEdge(msg *lnwire.ChannelUpdate,
+	pubKey *btcec.PublicKey) bool {
+
+	// Query for the policy, if not found, then this update is not for
+	// additional edges.
+	policy := p.getAddtionalEdgePolicy(
+		pubKey, msg.ShortChannelID.ToUint64(),
+	)
+	if policy == nil {
+		return false
+	}
+
+	// Ignore outdated message and return true to indicate that this update
+	// is meant for private edge although we won't apply it.
+	lastUpdate := time.Unix(int64(msg.Timestamp), 0)
```

Comment:
We should do normal channel update validation here. So `routing.ValidateChannelUpdateAnn`.